### PR TITLE
Fixes bypass to Tech-p

### LINKED ISSUE #408: application already submitted check in technical project applications
At the moment you are able to create several applications for technical project by bypassing domain select form

### PR DESCRIPTION
**Pull request by SaanviKumar13**
### Description

Prevents bypass to technical projects if application exists.

closes #408

### CODE REVIEW

File: client/dashboard/app/routes/applications.technical-project.tsx [hunk 1]
```diff
@@ -14,12 +14,13 @@ import type {
 } from "@remix-run/node";
```
Comment (shawshankkumar) [inline client/dashboard/app/routes/applications.technical-project.tsx:14]: no need to put a full stop after commit names

File: client/dashboard/app/routes/applications.technical-project.tsx [hunk 2]
```diff
@@ -35,9 +36,16 @@ export const meta: V2_MetaFunction = () => {
 
 export const links = () => [...projectTilesLinks(), ...projectLinks()];
 
-export const loader: LoaderFunction = ({ request }) => {
+export const loader: LoaderFunction = async ({ request }) => {
   const url = new URL(request.url);
   const slug = url.searchParams.get("slug");
+  const userDetails = await getUserDetails(request);
```
Comment (shawshankkumar) [inline client/dashboard/app/routes/applications.technical-project.tsx:42]: does this check for technicalp specifically? or any domain. Can you check what happens when you have filled the form for events and try to fill techp

**Review by shawshankkumar — CHANGES_REQUESTED** [review]

File: client/dashboard/app/routes/applications.technical-project.tsx [hunk 3]
```diff
@@ -35,9 +36,16 @@ export const meta: V2_MetaFunction = () => {
...
+  if (userDetails.application && userDetails.application.length > 0) {
+    const applicationExists = userDetails.application[0];
+    if (!(applicationExists.status === "draft")) {
```
Comment (shawshankkumar) [inline client/dashboard/app/routes/applications.technical-project.tsx:45]: also, i dont think its a good idea to use "draft" everywhere. create enum or create an object with as const In a shared folder and use it everywhere for things like pending, and other constant things.

File: client/dashboard/app/utils/constants.ts
Comment (sm-sami) [inline client/dashboard/app/utils/constants.ts:1]: move this to `types/index.ts`

File: client/dashboard/app/routes/applications.technical-project.tsx [hunk 4]
```diff
@@ -35,9 +37,14 @@ export const meta: V2_MetaFunction = () => {
 
 export const links = () => [...projectTilesLinks(), ...projectLinks()];
 
-export const loader: LoaderFunction = ({ request }) => {
+export const loader: LoaderFunction = async ({ request }) => {
   const url = new URL(request.url);
   const slug = url.searchParams.get("slug");
+  const userDetails = await getUserDetails(request);
```
Comment (sm-sami) [inline client/dashboard/app/routes/applications.technical-project.tsx:43]: Do the destructing here itself

**Review by sm-sami — CHANGES_REQUESTED** [review]

**Review by sm-sami — APPROVED** [review]

**Pull request merged**